cpp/src/io/text/multibyte_split.cu

```diff
@@ -0,0 +1,401 @@
+
+  __syncthreads();  // required before temp_memory re-use
+  auto prefix_callback =
+    OffsetScanCallback(temp_storage.offset_scan_callback, tile_output_offsets, tile_idx);
+
+  OffsetScan(temp_storage.offset_scan)
+    .ExclusiveSum(thread_offsets, thread_offsets, prefix_callback);
+
+  // Step 5: Assign outputs from each thread using match offsets.
+
+  if (abs_output_chars.size() > 0) {
+    for (int32_t i = 0; i < ITEMS_PER_THREAD and i < thread_input_size; i++) {
+      abs_output_chars[chunk_input_offset + thread_input_offset + i] = thread_chars[i];
+    }
+  }
+
+  if (abs_output_delimiter_offsets.size() > 0) {
+    for (int32_t i = 0; i < ITEMS_PER_THREAD and i < thread_input_size; i++) {
+      if (trie.get_match_length(thread_states[i]) > 0) {
+        auto const match_end = base_tile_idx * ITEMS_PER_TILE + thread_input_offset + i + 1;
+        abs_output_delimiter_offsets[thread_offsets[i]] = match_end;
+      }
+    }
+  }
+}
+
+}  // namespace
+
+namespace cudf {
+namespace io {
+namespace text {
+namespace detail {
+
+template <typename T>
+std::unique_ptr<column> create_column(rmm::device_uvector<T>&& values)
+{
+  auto size  = values.size();
+  auto dtype = cudf::data_type{cudf::type_to_id<T>()};
+
+  CUDF_EXPECTS(dtype.id() != type_id::EMPTY, "column type_id cannot be EMPTY");
+
+  return std::make_unique<cudf::column>(dtype, size, values.release(), rmm::device_buffer(), 0);
+}
+
+std::unique_ptr<column> create_char_column(rmm::device_uvector<char>&& values)
+{
+  auto size  = values.size();
+  auto dtype = cudf::data_type{type_id::INT8};
+
+  return std::make_unique<cudf::column>(dtype, size, values.release(), rmm::device_buffer(), 0);
+}
+
+std::unique_ptr<column> create_strings_column(rmm::device_uvector<char>&& chars,
+                                              rmm::device_uvector<int32_t>&& offsets,
+                                              rmm::cuda_stream_view stream,
+                                              rmm::mr::device_memory_resource* mr)
+{
+  auto num_strings    = offsets.size() - 1;
+  auto chars_column   = create_char_column(std::move(chars));
+  auto offsets_column = create_column(std::move(offsets));
+
+  return cudf::make_strings_column(
+    num_strings, std::move(offsets_column), std::move(chars_column), 0, {}, stream, mr);
+}
+
+void fork_stream(std::vector<rmm::cuda_stream_view> streams, rmm::cuda_stream_view stream)
+{
+  cudaEvent_t event;
+  cudaEventCreate(&event);
+  cudaEventRecord(event, stream);
+  for (uint32_t i = 0; i < streams.size(); i++) {
+    cudaStreamWaitEvent(streams[i], event, 0);
+  }
+  cudaEventDestroy(event);
+}
+
+void join_stream(std::vector<rmm::cuda_stream_view> streams, rmm::cuda_stream_view stream)
+{
+  cudaEvent_t event;
+  cudaEventCreate(&event);
+  for (uint32_t i = 0; i < streams.size(); i++) {
+    cudaEventRecord(event, streams[i]);
+    cudaStreamWaitEvent(stream, event, 0);
+  }
+  cudaEventDestroy(event);
+}
```

Comment:
what do you think about moving these to a util file?

Reply:
Since it's the first use case, I say we keep it here. When there's a second use case, we move it to a common location.